engine-tests/src/tests/promise_results_precompile.rs

```diff
@@ -119,14 +119,15 @@ fn test_promise_result_gas_cost() {
 
     let total_gas1 = y1 + baseline.all_gas();
     let total_gas2 = y2 + baseline.all_gas();
+
     assert!(
-        utils::within_x_percent(15, evm1, total_gas1 / NEAR_GAS_PER_EVM),
+        utils::within_x_percent(17, evm1, total_gas1 / NEAR_GAS_PER_EVM),
```

Comment:
I'm curious about why this seemingly small change has an impact on gas cost. Is the cost of reading that one key really so high? I wonder if it makes sense to go through `self.contract_code_cache` since then at least we are only reading the key once (assuming it is already being ready elsewhere in the sputnikVM code).

Reply:
Unfortunately yes, it's related to storage read. It's not a part of SputnikVM, as it's a stage of transaction validation and not part of an Execution Layer.

@birchmd do you probably remember, why we do not validate transactions according to `eip-3607`? As it's 3 years old EIP. Or we just missed it?

Actually, I discovered that issue when it was tested `SpuntikVM` with all ethereum-json tests. Because current tests for SputnikVM skip a lot of tests, including `EIP-3607`.

Reply:
Yes I'm sure it was just an oversight that EIP-3607 was not included earlier.

I still think it is worth trying recording the read of the code into the cache just in case SputnikVM also tries to read the same key later.

Reply:
@birchmd as you proposed added to `Engine` cache request.

Reply:
Can this value be lowered back to 15 with that change?

Reply:
Thanks for double checking. It makes me wonder if we should revisit the `NEAR_GAS_PER_EVM` constant which this test is meant to check is accurate. A 17% deviation is a far cry from the 5% deviation this test started with. I suppose that can be done as a separate PR though.